web/app/js/components/ResourceDetail.jsx

```diff
@@ -53,67 +47,131 @@ export class ResourceDetailBase extends React.Component {
     return {
       namespace: resource.namespace,
       resourceName: resource.name,
-      resourceType: resource.type
+      resourceType: resource.type,
+      resource,
+      pollingInterval: 2000,
+      resourceMetrics: [],
+      podMetrics: [],
+      pendingRequests: false,
+      loaded: false,
+      error: null
     };
   }
 
-  banner = () => {
-    const {error} = this.props;
+  componentDidMount() {
+    this.loadFromServer();
+    this.timerId = window.setInterval(this.loadFromServer, this.state.pollingInterval);
+  }
 
-    if (!error) {
+  componentWillReceiveProps(newProps) {
+    // React won't unmount this component when switching resource pages so we need to clear state
+    this.api.cancelCurrentRequests();
+    this.setState(this.getInitialState(newProps.match, newProps.pathPrefix));
+  }
+
+  componentWillUnmount() {
+    window.clearInterval(this.timerId);
+    this.api.cancelCurrentRequests();
+  }
+
+  loadFromServer() {
+    if (this.state.pendingRequests) {
+      return; // don't make more requests if the ones we sent haven't completed
+    }
+    this.setState({ pendingRequests: true });
+
+    let { resource } = this.state;
+    this.api.setCurrentRequests([
+      // inbound stats for this resource
+      this.api.fetchMetrics(
+        `${this.api.urlsForResource(resource.type, resource.namespace)}&resource_name=${resource.name}`
+      ),
+      // pod metrics for outbound stats --to this resource
+      // only gets traffic originating within the mesh
+      this.api.fetchMetrics(
+        `${this.api.urlsForResource("pod", resource.namespace)}&to_name=${resource.name}&to_type=${resource.type}`
```

Comment:
This query lists all pods that call the current resource, rather than all pods belonging to the resource itself. For instance, in linkerd namespace, I see:

![image](https://user-images.githubusercontent.com/9226/44366304-1b445680-a481-11e8-8854-0e1d9a28b9fe.png)

In this screenshot, the deployment is "prometheus", but the pods listed in the pods table belong to the "controller" and "grafana" deployments, not the "prometheus" deployment.

Unfortunately, I don't think there's a way using our existing stat APIs to get the data that we want to display in this table. This API call triggers the following query:

```
sum(increase(response_total{direction="outbound", dst_deployment="prometheus", dst_namespace="linkerd", namespace="linkerd"}[1m])) by (namespace, pod, classification, tls) 
```

But rather than grouping by (src) namespace and pod, we want to group by dst_namespace and dst_pod:

```
sum(increase(response_total{direction="outbound", dst_deployment="prometheus", dst_namespace="linkerd", namespace="linkerd"}[1m])) by (dst_namespace, dst_pod, classification, tls) 
```

So I think we'll have to try one of the other approaches outline in #1467 to get this data.